packages/sit-onyx/src/components/OnyxSwitch/OnyxSwitch.vue

```diff
@@ -0,0 +1,274 @@
+<script lang="ts" setup>
+import { OnyxIcon } from "@/index";
+import checkSmall from "@sit-onyx/icons/check-small.svg?raw";
+import xSmall from "@sit-onyx/icons/x-small.svg?raw";
+import type { OnyxSwitchProps } from "./types";
+import { computed } from "vue";
+const props = withDefaults(defineProps<OnyxSwitchProps>(), {
+  modelValue: false,
+  label: "Onyx Switch",
+  disabled: false,
+  required: false,
+  readonly: false,
+  invalid: false,
+});
+
+// TODO: Implement invalid disabled state.
+
+const emit = defineEmits<{
+  /** Emitted when the checked state changes. */
+  "update:modelValue": [value: boolean];
+}>();
+
+const isChecked = computed({
+  get: () => props.modelValue,
+  set: (value) => {
+    emit("update:modelValue", value);
+  },
+});
+</script>
+
+<template>
+  <label
+    tabindex="0"
+    class="onyx-switch"
+    :class="{ 'onyx-required-marker': props.required, 'onyx-optional-marker': !props.required }"
+  >
+    <input
+      v-model="isChecked"
+      class="onyx-switch__input"
+      :class="{
+        'onyx-switch__input--invalid': props.invalid,
+        'onyx-switch__input--readonly': props.readonly,
+        'onyx-switch__input--disabled': props.disabled,
+      }"
+      :aria-label="props.hideLabel ? props.label : undefined"
+      type="checkbox"
+      :disabled="props.disabled || props.readonly"
+      :required="props.required"
+    />
+    <span class="onyx-switch__container">
+      <span class="onyx-switch__track"></span>
+      <span class="onyx-switch__position">
+        <OnyxIcon class="onyx-switch__icon" :icon="isChecked ? checkSmall : xSmall" size="24px" />
+      </span>
+    </span>
+    <span v-if="props.label && !props.hideLabel" class="onyx-switch__label">{{ props.label }}</span>
```

Comment:
When label is mandatory:

```suggestion
    <span v-if="!props.hideLabel" class="onyx-switch__label">{{ props.label }}</span>
```